src/main/java/org/spongepowered/api/data/type/RailDirection.java

```diff
@@ -24,11 +24,103 @@
  */
 package org.spongepowered.api.data.type;
 
+import static com.google.common.base.Preconditions.checkNotNull;
+
 import org.spongepowered.api.CatalogType;
+import org.spongepowered.api.Sponge;
 import org.spongepowered.api.util.Cycleable;
+import org.spongepowered.api.util.Direction;
 import org.spongepowered.api.util.annotation.CatalogedBy;
 
+import java.util.Optional;
+
 @CatalogedBy(RailDirections.class)
 public interface RailDirection extends CatalogType, Cycleable<RailDirection> {
```

Comment:
This class needs a javadoc explaining what it represents. I'm also not sure about the name `RailDirection` because it doesn't really represent the direction of the rails. It represents which edge the two rail piece termini are at, not really the direction of the rail itself.

Would like others' input on this.